src/git-pull-request

```diff
@@ -2,25 +2,16 @@
 source "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/helpers/helpers.sh"
 
 
-# Returns the source for a new Bitbucket pull request
-function bitbucket_source {
-  sha=$(git log --format="%H" -1 | cut -c-12)
-  echo "$repository:$sha:$branch" | sed 's/\//%2F/g' | sed 's/\:/%3A/g'
+function preconditions {
+  activate_drivers 'code_hosting' \
+                   "$(remote_domain)" \
+                   'Pull requests can only be created on Bitbucket and GitHub'
```

Comment:
I spoke too soon. The error messages are command specific, so I keep them in the commands. I moved all other initialization data into the driver family though.